### ISSUE TITLE
array multiline allman style

### ISSUE THREAD

**Issue by TheElegantCoding** [eslint-stylistic/eslint-stylistic#170]
Hello, is there any rule for putting the array bracket to a new line if it is multiline? like Allman style

for example:

First bracket error:

```ts
  const array = [
    'props', 
    { hello: 'world' }
  ]
```
correct method

```ts
  const array = 
  [
    'props', 
    { hello: 'world' }
  ]
```

also these dont work with object 

error
```ts
const stylisticDisabledRule = {
  'jsx-sort-props': 'off'
};

const array = [
  'error',
  {
    hello: {
      good: "morning"
    }
  }
]
```
expected correction
```ts
const stylisticDisabledRule = 
{
  'jsx-sort-props': 'off'
};

const array = 
[
  'error',
  {
    hello: 
    {
      good: "morning"
    }
  }
]
```

Edit: 

a long time ago in the eslint team, they proposed adding `object-curly-style` and `array-bracket-style` as new rules for these cases in this issue [#5203](https://github.com/eslint/eslint/issues/5203), but looks like this never implements

**Comment by ljharb** [comment]
allman style leads to bugs; it should be avoided, and only OTBS should be used in JS.

**Comment by kecrily** [comment]
> allman style leads to bugs; it should be avoided, and only OTBS should be used in JS.

For some additional information on this, use allman style in the `return` statement. semicolon is automatically inserted after `return`[^1].

[^1]: [Rules of Automatic Semicolon Insertion](https://262.ecma-international.org/5.1/#sec-7.9.1)

**Comment by antfu** [comment]
I'd suggest implementing it and distributing it as your own plugin. If you are able to prove this is something has a certain amount of demand, with a solid implementation, then we could discuss whether to have it in this project. Thanks for bringing this up.
